src/core/education/service/education.service.ts

```diff
@@ -0,0 +1,138 @@
+import {
+  BadRequestException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+import { CreateEducationDto } from '../dto/create-education.dto';
+import { UpdateEducationDto } from '../dto/update-education.dto';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Education } from '../entities/education.entity';
+import { Repository } from 'typeorm';
+import { UserActiveInterface } from 'src/common/interface/user-active-interface';
+import {
+  EDUCATION_NOT_FOUND,
+  ERROR_EDUCATION_IS_UCAB,
+  ERROR_EDUCATION_PRINCIPAL_ALREADY_EXISTS,
+} from '../message';
+
+@Injectable()
+export class EducationService {
+  constructor(
+    @InjectRepository(Education)
+    private educationRepository: Repository<Education>,
+  ) {}
+
+  async create(
+    createEducationDto: CreateEducationDto,
+    user: UserActiveInterface,
+  ) {
+    const newEducation = await this.educationRepository.save({
+      ...createEducationDto,
+      profileId: user.id,
+    });
+
+    return newEducation;
+  }
+
+  async findAll(idProfile: number) {
+    const educations = await this.educationRepository.find({
+      where: {
+        deleteAt: null,
+        profileId: idProfile,
+      },
+    });
+
+    if (educations.length === 0) {
+      throw new NotFoundException(EDUCATION_NOT_FOUND);
+    }
+
+    return educations;
+  }
+
+  async findOne(id: number) {
+    const education = await this.educationRepository.findOne({
+      where: {
+        id,
+      },
+    });
+
+    if (!education) {
+      throw new NotFoundException(EDUCATION_NOT_FOUND);
+    }
+
+    return education;
+  }
+
+  async update(
+    id: number,
+    updateEducationDto: UpdateEducationDto,
+    user: UserActiveInterface,
+  ): Promise<void> {
+    let education;
+
+    const validateEducationIsUCAB = await this.educationRepository.findOne({
+      where: {
+        profileId: user.id,
+        id: id,
+      },
+    });
+
+    if (
+      validateEducationIsUCAB.isUCAB === true &&
+      (updateEducationDto.principal === true ||
+        updateEducationDto.principal === false)
```

Comment:
![image](https://github.com/EquestriaReapers/LRT-Backend/assets/22270579/b8cebf08-b106-4f53-bbd8-a89f3a4232b1)

Papa xd `q || !q = true`

```suggestion
      validateEducationIsUCAB.isUCAB === true
```

Reply:
Por cierto, no hace falta el === true si es que es true

```suggestion
      validateEducationIsUCAB.isUCAB
```

Reply:
Comentasinho no es un if para que solo sea vdd es para verificar que no entre un undefined